File: distribution/python/dqops/client/models/label_model.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="LabelModel")


@_attrs_define
class LabelModel:
    """Label model that is returned by the REST API. A label is a tag that was assigned to a data source, table, column or
    a single check. Labels play the role of a business glossary.

        Attributes:
            label (Union[Unset, str]): Label text.
            labels_count (Union[Unset, int]): The number of data assets tagged with this label.
            nested_labels_count (Union[Unset, int]): The number of data assets tagged with nested labels below this prefix
                node. For example, if the current label is "address", and there are nested labels "address/city" and
                "address/zipcode", this value returns the count of data assets tagged with these nested tags.
    """

    label: Union[Unset, str] = UNSET
    labels_count: Union[Unset, int] = UNSET
    nested_labels_count: Union[Unset, int] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        label = self.label
        labels_count = self.labels_count
        nested_labels_count = self.nested_labels_count

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if label is not UNSET:
            field_dict["label"] = label
        if labels_count is not UNSET:
            field_dict["labels_count"] = labels_count
        if nested_labels_count is not UNSET:
            field_dict["nested_labels_count"] = nested_labels_count

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        label = d.pop("label", UNSET)

        labels_count = d.pop("labels_count", UNSET)

        nested_labels_count = d.pop("nested_labels_count", UNSET)

        label_model = cls(
            label=label,
            labels_count=labels_count,
            nested_labels_count=nested_labels_count,
        )

        label_model.additional_properties = d
        return label_model

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/rest/models/metadata/LabelModel.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.rest.models.metadata;

import com.dqops.metadata.labels.labelcontainers.LabelCounter;
import com.dqops.utils.docs.generators.SampleStringsRegistry;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;


/**
 * Label model that is returned by the REST API. A label is a tag that was assigned to a data source, table, column or a single check.
 * Labels play the role of a business glossary.
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@ApiModel(value = "LabelModel", description = "Label model that is returned by the REST API. A label is a tag that was assigned to a data source, table, column or a single check. Labels play the role of a business glossary.")
public class LabelModel {
    /**
     * Label text.
     */
    @JsonPropertyDescription("Label text.")
    private String label;

    /**
     * The number of data assets tagged with this label.
     */
    @JsonPropertyDescription("The number of data assets tagged with this label.")
    private int labelsCount;

    /**
     * The number of data assets tagged with nested labels below this prefix node. For example, if the current label is "address", and there are nested labels "address/city" and "address/zipcode",
     * this value returns the count of data assets tagged with these nested tags.
     */
    @JsonPropertyDescription("The number of data assets tagged with nested labels below this prefix node. For example, if the current label is \"address\", " +
            "and there are nested labels \"address/city\" and \"address/zipcode\", this value returns the count of data assets tagged with these nested tags.")
    private int nestedLabelsCount;


    public LabelModel() {
    }

    public LabelModel(String label, int labelsCount, int nestedLabelsCount) {
        this.label = label;
        this.labelsCount = labelsCount;
        this.nestedLabelsCount = nestedLabelsCount;
    }

    /**
     * Creates a label model from a label counter.
     * @param labelCounter Label counter.
     * @return Label model.
     */
    public static LabelModel fromLabelCounter(LabelCounter labelCounter) {
        return new LabelModel(labelCounter.getLabel(), labelCounter.getLabelsCount(), labelCounter.getNestedLabelsCount());
    }

    public static class LabelModelSampleFactory implements SampleValueFactory<LabelModel> {
        @Override
        public LabelModel createSample() {
            return new LabelModel() {{
                setLabel(SampleStringsRegistry.getLabelSample());
                setLabelsCount(10);
                setNestedLabelsCount(2);
            }};
        }
    }
}
